framework/src/base/NonlinearSystem.C

```diff
@@ -680,16 +694,28 @@ NonlinearSystem::computeResidual(NumericVector<Number> & residual, Moose::Kernel
 
   try
   {
-    residual.zero();
```

Comment:
You could have other terms that are not TimeKernels that still involve the time derivative of variables.  See, for instance, `CoupledTimeDerivative`: https://github.com/idaholab/moose/blob/devel/framework/include/kernels/CoupledTimeDerivative.h